api_v3/lib/KalturaLiveEntryService.php

```diff
@@ -319,9 +319,12 @@ private function createRecordedEntry(LiveEntry $dbEntry, $mediaServerIndex)
 	    // If while we were waiting for the lock, someone has updated the recorded entry id - we should use it.
 	    $dbEntry->reload();
 	    if(($dbEntry->getRecordStatus() != RecordStatus::PER_SESSION) && ($dbEntry->getRecordedEntryId())) {
-	    	$lock->unlock();
 	    	$recordedEntry = entryPeer::retrieveByPK($dbEntry->getRecordedEntryId());
-	    	return $recordedEntry;
+	    	if($recordedEntry)
+	    	{
+	    		$lock->unlock();
+	    		return $recordedEntry;
+	    	}	    	
 	    }
 
```

Comment:
The entire reason for this change is so we would validate if the recordedEntry really exists before freeing the lock and not continuing to the flow that re-creates the recordedEntry